code/src/main/kotlin/com/expediagroup/sdk/lodgingconnectivity/graphql/BaseGraphQLClient.kt

```diff
@@ -16,78 +16,106 @@
+import com.expediagroup.sdk.lodgingconnectivity.configuration.ClientConfiguration
+import java.util.concurrent.CompletableFuture
 
-/**
- * An internal base implementation of a GraphQL client for executing GraphQL queries, mutations, and subscriptions.
- *
- * This class integrates the Apollo GraphQL client with a custom `ExpediaGroupClient` for handling HTTP communication
- * and error management. It provides a foundation for more specific client implementations by executing operations
- * with error handling.
- *
- * @param config The configuration for the `ExpediaGroupClient`
- */
-internal class BaseGraphQLClient(config: ExpediaGroupClientConfiguration) : GraphQLExecutor {
+class BaseGraphQLClient(configuration: FullClientConfiguration) : GraphQLExecutor {
+    private val engine: ApiClientApolloHttpEngine = ApiClientApolloHttpEngine(
+        createApiClient(
+            configuration = configuration
+        )
+    )
 
-    // Custom client for handling HTTP requests and responses.
-    private val expediaGroupClient =
-        object : ExpediaGroupClient(clientConfiguration = config, namespace = "lodging-connectivity-sdk") {
-            override suspend fun throwServiceException(response: HttpResponse, operationId: String) {
-                throw ExpediaGroupServiceException("Service error occurred for operation $operationId.\nResponse: $response")
-            }
-        }
+    companion object {
+        @JvmStatic
+        fun builder() = ClientConfiguration.builder()
+    }
 
     private val apolloClient: ApolloClient = ApolloClient.Builder()
-        .serverUrl(config.endpoint!!)
-        .httpEngine(KtorHttpEngine(expediaGroupClient.httpClient))
+        .serverUrl(configuration.getEndpoint())
+        .httpEngine(engine)
         .build()
 
+    class Builder() : DefaultClientBuilder<BaseGraphQLClient>() {
+        override fun build(): BaseGraphQLClient {
+            return BaseGraphQLClient(this.buildConfiguration())
+        }
+    }
+
     /**
      * Executes a GraphQL query and returns the result.
      *
      * @param query The GraphQL query to execute.
      * @return The result of the query execution, with errors handled.
      * @throws ExpediaGroupServiceException If the query execution returns errors.
      */
-    override fun <T : Query.Data> execute(query: Query<T>): T {
-        return runBlocking {
-            apolloClient.query(query).execute().apply {
-                if (exception != null) {
-                    throw ExpediaGroupServiceException(exception?.message)
-                }
-                if (hasErrors()) {
-                    throw ExpediaGroupServiceException(errors.toString())
+    override fun <T : Query.Data> executeAsync(query: Query<T>): CompletableFuture<T> {
+        val promise: CompletableFuture<T> = CompletableFuture()
+
+        apolloClient.query(query).enqueue { response ->
+            try {
+                if (response.hasErrors()) {
```

Comment:
We will have tons of conflicts once we want to merge it into main 🥲 This logic for error handling has changed